### ISSUE TITLE
Investigate certificate renewal

### ISSUE THREAD

**Issue by mariocimet** [bravetechnologycoop/BraveButtons#11]

**Comment by schwarrrtz** [comment]
ok so i've done a bit of investigation into the chatbot logs, and it seems like in at least one instance (around 26 Aug 2019) the certbot renew command ran without the Node server process being restarted. this would mean that the new certificate doesn't end up getting used by the server until something else causes pm2 to restart the Node process.

**Comment by schwarrrtz** [comment]
there is a pre-hook and a post-hook on the certbot command that are supposed to cause pm2 to restart the Node process once the cert renewal is complete (see `chatbot/setup_server.sh` for details). I'm wondering if maybe these aren't working, possibly because cron executes the command from a different folder or something?

**Comment by mariocimet** [comment]
Sampath and I just saw the same issue on odetect-dev2. The certificate had been renewed but the server had not been restarted. 

We can probably guard against this by chaining the commands we schedule - i.e. when we schedule certbot renew it should also do sudo pm2 restart

**Comment by mariocimet** [comment]
Sampath asks: is there a reason why we don't do something like:

```
echo "0 0 * * 0 pm2 stop server" > crontab.tmp
echo "0 0 * * 0 certbot renew" > crontab.tmp
echo "0 0 * * 0 pm2 start server" > crontab.tmp
```

**Comment by sampathsatti** [comment]
So did some sleuthing on odetect-dev2

- Check history of when certificates were issued (https://transparencyreport.google.com/https/certificates) 
- Reveals #1 from Jun 26 to Sep 4, #2 from Aug 25 to Nov 23, #3 from Oct 25 to Jan 22
- Reveals certbot renew is working as expected (renew before 30 days of certificate expiry)

-Then went to check when pm2 was restarted - last seems to be Aug 21.
-This indicates that prehook (pm2 stop) and posthook (pm2 start) on the renew command failed 

To test this hypothesis again - modified certbot to force renew the certificate everytime crontab runs. Ideally when it does it this midnight we should see a pm2 log as well. If we don't, we'll need to diagnose this hook syntax more or chain the cron commands with pm2 stop, certbot renew, pm2 start

**Comment by schwarrrtz** [comment]
i don't think that chaining the pm2 commands in cron will help - after some testing, i'm pretty sure the issue occurs because we're adding the commands to root's crontab, meaning that they get executed with root's home directory as the working directory. but the pm2 config for our servers is usually stored in the home directory of the 'brave' user, or whatever the non-root admin user's name is

**Comment by schwarrrtz** [comment]
pretty sure this is fixed in #20.

**Comment by mariocimet** [comment]
sweet! reviewed and merged
